ess/src/API/EMBC.ESS/Resources/Suppliers/Mappings.cs

```diff
@@ -27,15 +27,15 @@ public Mappings()
                 .ForPath(d => d.Contact.LastName, opts => opts.MapFrom(s => s.era_PrimaryContact != null ? s.era_PrimaryContact.era_lastname : null))
                 .ForPath(d => d.Contact.Phone, opts => opts.MapFrom(s => s.era_PrimaryContact != null ? s.era_PrimaryContact.era_contactnumber : null))
                 .ForPath(d => d.Contact.Email, opts => opts.MapFrom(s => s.era_PrimaryContact != null ? s.era_PrimaryContact.emailaddress : null))
-                .ForMember(d => d.Team, opts => opts.MapFrom(s => s.era_era_supplier_era_essteamsupplier_SupplierId.SingleOrDefault(ts => ts.era_isprimarysupplier == true)))
+                .ForMember(d => d.PrimaryTeams, opts => opts.MapFrom(s => s.era_era_supplier_era_essteamsupplier_SupplierId.Where(ts => ts.era_isprimarysupplier == true)))
                 .ForMember(d => d.SharedWithTeams, opts => opts.MapFrom(s => s.era_era_supplier_era_essteamsupplier_SupplierId.Where(ts => ts.era_isprimarysupplier != true)))
                 .AfterMap((s, d) =>
                 {
-                    var responsibleTeam = s.era_era_supplier_era_essteamsupplier_SupplierId.SingleOrDefault(ts => ts.era_isprimarysupplier == true);
-                    if (responsibleTeam == null)
+                    var responsibleTeams = s.era_era_supplier_era_essteamsupplier_SupplierId.Where(ts => ts.era_isprimarysupplier == true);
+                    if (responsibleTeams == null)
```

Comment:
Any() would probably work as it will never be null

Reply:
I'm using the found teams to determine if the Status should be Active or Inactive.

Reply:
I understand, still, responsibleTeam will never be null as Where will always return an enumerable, never null